### ISSUE TITLE
Incompatible with Java 16 due to xodus-dnq

### ISSUE THREAD

**Issue by mooziii** [bluefireoly/pacmc#3]
Hi, im getting this error when running `pacmc list` or `pacmc install <mod>`
Im on manjaro linux with java 16

```
Exception in thread "main" java.lang.ExceptionInInitializerError
	at javassist.util.proxy.DefineClassHelper.<clinit>(DefineClassHelper.java:216)
	at javassist.util.proxy.FactoryHelper.toClass(FactoryHelper.java:128)
	at javassist.util.proxy.ProxyFactory.createClass3(ProxyFactory.java:552)
	at javassist.util.proxy.ProxyFactory.createClass1(ProxyFactory.java:475)
	at javassist.util.proxy.ProxyFactory.createClass(ProxyFactory.java:444)
	at javassist.util.proxy.ProxyFactory.create(ProxyFactory.java:721)
	at kotlinx.dnq.util.XdHierarchyNode$naturalPersistentClassInstance$2.invoke(XdHierarchyNode.kt:96)
	at kotlinx.dnq.util.XdHierarchyNode$naturalPersistentClassInstance$2.invoke(XdHierarchyNode.kt:33)
	at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
	at kotlinx.dnq.util.XdHierarchyNode.getNaturalPersistentClassInstance(XdHierarchyNode.kt)
	at kotlinx.dnq.util.DNQMetaDataUtilKt.initMetaData(DNQMetaDataUtil.kt:46)
	at net.axay.pacmc.storage.Xodus.<clinit>(Xodus.kt:28)
	at net.axay.pacmc.commands.Archive$List.run(Archive.kt:54)
	at com.github.ajalt.clikt.parsers.Parser.parse(Parser.kt:204)
	at com.github.ajalt.clikt.parsers.Parser.parse(Parser.kt:213)
	at com.github.ajalt.clikt.parsers.Parser.parse(Parser.kt:213)
	at com.github.ajalt.clikt.parsers.Parser.parse(Parser.kt:17)
	at com.github.ajalt.clikt.core.CliktCommand.parse(CliktCommand.kt:396)
	at com.github.ajalt.clikt.core.CliktCommand.parse$default(CliktCommand.kt:393)
	at com.github.ajalt.clikt.core.CliktCommand.main(CliktCommand.kt:411)
	at com.github.ajalt.clikt.core.CliktCommand.main(CliktCommand.kt:436)
	at net.axay.pacmc.ManagerKt.main(Manager.kt:10)
Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make protected final java.lang.Class java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain) throws java.lang.ClassFormatError accessible: module java.base does not "opens java.lang" to unnamed module @5fb2de77
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:357)
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:297)
	at java.base/java.lang.reflect.Method.checkCanSetAccessible(Method.java:199)
	at java.base/java.lang.reflect.Method.setAccessible(Method.java:193)
	at javassist.util.proxy.SecurityActions$3.run(SecurityActions.java:94)
	at javassist.util.proxy.SecurityActions$3.run(SecurityActions.java:90)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:554)
	at javassist.util.proxy.SecurityActions.getMethodHandle(SecurityActions.java:89)
	at javassist.util.proxy.DefineClassHelper$SecuredPrivileged$2.getDefineClassMethodHandle(DefineClassHelper.java:143)
	at javassist.util.proxy.DefineClassHelper$SecuredPrivileged$2.<init>(DefineClassHelper.java:136)
	at javassist.util.proxy.DefineClassHelper$SecuredPrivileged.<clinit>(DefineClassHelper.java:134)
	... 22 more````

**Comment by bluefireoly** [comment]
Yes it occurs if you are trying to run `pacmc` with Java 16.

This is an issue with `xodus-dnq`, I have already created an issue there a while ago: https://github.com/JetBrains/xodus-dnq/issues/75

For now I suggest to run pacmc with Java 11.

**Comment by bluefireoly** [comment]
>  Im on manjaro linux with java 16

If you have installed multiple Java versions, you can switch between them like this:
https://wiki.archlinux.org/title/Java#Switching_between_JVM

**Comment by mooziii** [comment]
Thanks, it works now with java 11

**Comment by bluefireoly** [comment]
Fixed with version **0.2.0**.

@mooziii for you that means that you can just update pacmc from the AUR
